httpclient/keepalive_syscall_linux_test.go

```diff
@@ -29,6 +30,7 @@ var _ = Describe("Linux-specific tests", func() {
 
 			ln, err := net.Listen("tcp", laddr)
 			Expect(err).ToNot(HaveOccurred())
+			time.Sleep(3 * time.Second)
```

Comment:
Curious if there's something like an "on ready" callback we could pass to `Listen()`

Reply:
@aramprice that sounds like the ideal solution, the documentation for Listen() didn't mention any sort of mechanism to verify that the port was opened. I think it's assumed it's a synchronous method. `Eventually` sounds like a good workaround. Thanks